integration/src/wallets/tally.ts

```diff
@@ -0,0 +1,166 @@
+import * as core from "@shapeshiftoss/hdwallet-core";
+import * as tally from "@shapeshiftoss/hdwallet-tally";
+
+export function name(): string {
+  return "Tally";
+}
+
+export function createInfo(): core.HDWalletInfo {
+  return new tally.TallyHDWalletInfo();
+}
+
+export async function createWallet(): Promise<core.HDWallet> {
+  const provider = {
+    request: jest.fn(({ method, params }: any) => {
+      switch (method) {
+        case "eth_accounts":
+          return ["0x3f2329C9ADFbcCd9A84f52c906E936A42dA18CB8"];
+        case "personal_sign": {
+          const [message] = params;
+
+          if (message === "48656c6c6f20576f726c64")
+            return "0x29f7212ecc1c76cea81174af267b67506f754ea8c73f144afa900a0d85b24b21319621aeb062903e856352f38305710190869c3ce5a1425d65ef4fa558d0fc251b";
+
+          throw new Error("unknown message");
+        }
+        case "eth_sendTransaction": {
+          const [{ to }] = params;
+
+          return `txHash-${to}`;
+        }
+        default:
+          throw new Error(`ethereum: Unknown method ${method}`);
+      }
+    }),
+  };
+  const wallet = new tally.TallyHDWallet(provider);
+  await wallet.initialize();
+  return wallet;
+}
+
+export function selfTest(get: () => core.HDWallet): void {
+  let wallet: tally.TallyHDWallet;
+
+  beforeAll(async () => {
+    const w = get() as tally.TallyHDWallet;
+
+    if (tally.isTally(w) && !core.supportsBTC(w) && core.supportsETH(w)) {
+      wallet = w;
+    } else {
+      throw "Wallet is not a Tally";
+    }
+  });
+
+  it("supports Ethereum mainnet", async () => {
+    if (!wallet) return;
+    expect(await wallet.ethSupportsNetwork()).toEqual(true);
+  });
+
+  it("does not support BTC", async () => {
+    if (!wallet) return;
+    expect(core.supportsBTC(wallet)).toBe(false);
+  });
+
+  it("does not support Native ShapeShift", async () => {
+    if (!wallet) return;
+    expect(wallet.ethSupportsNativeShapeShift()).toEqual(false);
+  });
+
+  it("does not support EIP1559", async () => {
```

Comment:
Is this correct? Does Tally not have EIP-1559 support?

Reply:
That is not correct. Tally does support EIP-1559 and I'm adjusting to suit.